src/Codeception/Module/WebDriver.php

```diff
@@ -976,17 +978,39 @@ public function dontSeeLink($text, $url = null)
             }
             return;
         }
+
+        $expectedUrl = $this->getAbsoluteUrlFor($url);
         $nodes = array_filter(
             $nodes,
-            function (WebDriverElement $e) use ($url) {
-                return trim($e->getAttribute('href')) == trim($url);
+            function (WebDriverElement $e) use ($expectedUrl) {
+                $elementHref = $this->getAbsoluteUrlFor($e->getAttribute('href'));
+                return $elementHref === $expectedUrl;
             }
         );
         if (!empty($nodes)) {
             $this->fail("Link containing text '$text' and URL '$url' was found in page " . $this->_getCurrentUri());
         }
     }
 
+
+    /**
+     * Returns an absolute URL for the passed URI with the current URL
+     * as the base path.
+     *
+     * @param string $uri the absolute or relative URI
+     * @return string the absolute URL
+     */
+    private function getAbsoluteUrlFor($uri)
```

Comment:
The last line calls a method of `Codeception\Util\Uri`.
The first 2 lines are WebDriver specific. (It is funny that Selenium always returns an absolute url for href, but it can return a relative url for current uri).
The check for empty url should be moved to `Uri::mergeUrls`, but I have copied it from InnerBrowser and was too lazy to refactor/didn't want to break anything else.

There are similar methods in Guzzle and Guzzle6 connector too.